File: backend/routes/feedback.py
from fastapi import APIRouter
from models import Feedback, InteractionType
from services.kafka_service import KafkaService

router = APIRouter()


@router.post("/feedback", status_code=204)
def submit_feedback(payload: Feedback):
    KafkaService().send_interaction(
        payload.userId,
        payload.productId,
        InteractionType.RATE,
        payload.rating,
        review_title=payload.title,
        review_content=payload.comment,
    )
    return  # TODO add logic

File: frontend/src/services/feedback.ts
import { apiRequest, ServiceLogger } from './api';

export interface Feedback {
  userId: string;
  productId: string;
  rating: number;
  title?: string;
  comment?: string;
}

export const submitFeedback = async (feedback: Feedback): Promise<void> => {
  ServiceLogger.logServiceCall('submitFeedback', { feedback });
  return apiRequest<void>('/feedback', 'submitFeedback', {
    method: 'POST',
    body: feedback,
  });
};
